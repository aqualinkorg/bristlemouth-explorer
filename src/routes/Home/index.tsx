import {
  Button,
  Link,
  Stack,
  TextField,
  Typography,
  styled,
} from '@mui/material';
import bristlemouthLogo from 'src/assets/bristlemouth-logo.png';
import { spottersRequest } from 'src/store/spotters/spottersSlice';
import { useAppDispatch } from 'src/store/hooks';
import { bristlemouthURL, sofarDocsURL } from 'src/helpers/constants';
import Footer from 'src/common/Footer';
import { useNavigate } from 'react-router-dom';
import React from 'react';
import {
  setSettings,
  settingsSelector,
} from 'src/store/settings/settingsSlice';
import { useSelector } from 'react-redux';

const WrapperDiv = styled('div')(({ theme }) => ({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'flex-end',
  alignItems: 'center',
  paddingTop: theme.spacing(10),
  gap: theme.spacing(12),
}));

const Logo = styled('img')(({ theme }) => ({
  borderRadius: theme.spacing(1),
  height: '5rem',
  width: '5rem',
}));

const StyledLink = styled(Link)(() => ({
  color: 'black',
  textDecorationColor: 'unset',
}));

function Home() {
  const dispatch = useAppDispatch();
  const navigate = useNavigate();

  const { sofarApiToken } = useSelector(settingsSelector);

  const [token, setToken] = React.useState<string>(sofarApiToken || '');

  async function onTokenSubmit() {
    const result = await dispatch(spottersRequest({ token }));
    if (result.meta.requestStatus === 'rejected') return;

    dispatch(setSettings({ sofarApiToken: token }));
    navigate('/sensors');
  }

  const handleTokenChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setToken(e.target.value);
  };

  // Mask all characters except the last 6
  const maskedToken = token.replace(/.(?=.{6})/g, 'â€¢');

  return (
    <WrapperDiv>
      <Stack direction="column" alignItems="center" spacing={10}>
        <Link target="_blank" rel="noopener noreferrer" href={bristlemouthURL}>
          <Logo src={bristlemouthLogo} alt="Bristlemouth logo" />
        </Link>
        <Stack direction="column" alignItems="center" spacing={5}>
          <Typography variant="h1" fontWeight="bold">
            Bristlemouth Explorer
          </Typography>
          <Stack direction="column" gap="1rem" spacing={1} width="20rem">
            <Stack>
              <Typography fontWeight="bold">
                Enter your Sofar API Token
              </Typography>
              <TextField
                value={maskedToken}
                onChange={handleTokenChange}
                placeholder="XXXX-XXX-XXXX-XXX"
                variant="outlined"
                size="small"
              />
            </Stack>

            <Button
              color="primary"
              variant="contained"
              style={{
                color: 'white',
                fontWeight: 'bold',
                textTransform: 'none',
              }}
              onClick={() => onTokenSubmit()}
            >
              Go
            </Button>
          </Stack>
          <Stack direction="column" alignItems="center" spacing={1}>
            <StyledLink
              target="_blank"
              rel="noopener noreferrer"
              href="mailto:info@aqualink.org"
            >
              Can&apos;t find or need a Sofar API Token?
            </StyledLink>
            <StyledLink
              target="_blank"
              rel="noopener noreferrer"
              href={sofarDocsURL}
            >
              What can you do with this Sofar API?
            </StyledLink>
          </Stack>
        </Stack>
      </Stack>
      <Footer />
    </WrapperDiv>
  );
}

export default Home;
