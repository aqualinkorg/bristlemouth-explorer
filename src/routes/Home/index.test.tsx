import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import Home from '.';
import { BrowserRouter } from 'react-router-dom';

const mockStore = configureStore([]);

const store = mockStore({
  spotters: {
    list: [],
    spottersRequestLoading: false,
    error: null,
  },
});

store.dispatch = jest.fn();

it('renders as expected', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
