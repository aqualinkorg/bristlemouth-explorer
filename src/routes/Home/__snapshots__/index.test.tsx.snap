// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div>
    Home Page
  </div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1o4jfbn-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-1so0nez-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-3w66bp-MuiTypography-root"
        >
          Bristlemouth for developers
        </h5>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-93nype-MuiGrid-root"
          style="gap: 32px;"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-93nype-MuiGrid-root"
              style="gap: 8px;"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
              >
                Developed with ❤️ by
              </p>
              <img
                alt="Aqualink logo"
                class="css-4vnfkb"
                src="[object Object]"
              />
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-93nype-MuiGrid-root"
              style="gap: 16px;"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-zpphrz-MuiTypography-root"
              >
                Bristlemouth is an open 
                <br />
                 standard pioneered by
              </p>
              <img
                alt="Sofar logo"
                class="css-4vnfkb"
                src="[object Object]"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-hlnzui-MuiTypography-root"
        >
          © 
          2023
           All rights reserved 
          <br />
           Privacy policy - Contact Us
        </p>
      </div>
    </div>
  </header>
</div>
`;
